src/glove.jl

```diff
@@ -0,0 +1,74 @@
+                "617afb2fe6cbd085c235baf7a465b96f4112bd7f7ccb2b2cbd649fed9cbcf2fb",
+                ["50d", "100d", "200d", "300d"]),
+               ("glove.42B.300d",
+                "Trained on 42B tokens (Common Crawl), 1.9M vocab, uncased, Includes 300d vectors. 1.75 GB download.",
+                "03d5d7fa28e58762ace4b85fb71fe86a345ef0b5ff39f5390c14869da0fc1970",
+               []),
+               ("glove.840B.300d",
+                "Trained on 840B tokens (Common Crawl), 2.2M vocab, cased. Includes 300d vectors. 2.03 GB download.",
+                "c06db255e65095393609f19a4cfca20bf3a71e20cc53e892aafa490347e3849f",
+                []),
+               ("glove.twitter.27B",
+                "Trained on 2B tweets, 27B tokens, 1.2M vocab, uncased, 25d, 50d, 100d, & 200d vectors, 1.42 GB download.",
+                "792af52f795d1a32c9842a3240f5f3fe5e941a8ff6df5eb0f9d668092ebc019c",
+                ["25d", "50d", "100d", "200d"])]
+    for (depname, description, sha, dims) in vectors
+        register(DataDep(depname,
+                         """
+                         Pretrained GloVe word embeddings.
+                         Website: https://nlp.stanford.edu/projects/glove/
+                         Author: Jeffrey Pennington, Richard Socher, Christopher D. Manning 
+                         Year: 2014
+                         Licence: Open Data Commons Public Domain Dedication and License (PDDL)
+
+                         Paper:
+                             Jeffrey Pennington, Richard Socher, Christopher D. Manning. GloVe: Global Vectors for Word Representation. In Proceedings of EMNLP, 2014.
+
+                         $description
+                         """,
+                         "https://nlp.stanford.edu/data/$(depname).zip",
+                         sha,
+                         post_fetch_method = unpack))
+        if length(dims) >= 1
+            append!(language_files(GloVe{:en}), ["$(depname)/$(depname).$(dim).txt" for dim in dims])
+        else
+            push!(language_files(GloVe{:en}), "$depname/$depname.txt")
+        end
+    end
+end
+
+function _load_embeddings(::Type{<:GloVe}, embedding_file, max_vocab_size, keep_words)
+    local LL, indexed_words, index, D
+    if length(keep_words) > 0
+        max_vocab_size = length(keep_words)
+    end
+    D = open(embedding_file) do f
+        length(split(readline(f))) - 1
+    end
+    max_stored_vocab_size = min(max_vocab_size, glove_max_size)
+    indexed_words = Vector{String}(undef, max_stored_vocab_size)
+    LL = Array{Float32}(undef, D, max_stored_vocab_size)
+    open(embedding_file) do f
+        index = 1
+        for line in eachline(f)
+            xs = split(line)
+            word = xs[1]
+            if length(keep_words) == 0 || (word in keep_words)
+                index > max_stored_vocab_size && break
+                @inbounds LL[:, index] = parse.(Float32, xs[2:end]) # 2.64, then ~0.3 (?)
```

Comment:
What does  `# 2.64, then ~0.3 (?)` mean?

Reply:
Oh, oops! I was doing some experiments with loading the vectors a few different ways to avoid the size issue (including the vector of vectors then hcat at the end idea you mentioned below), and I was leaving little notes for myself to remember how long each one was taking. (in that case, it was ~2.64 seconds for the first call, then ~0.3 in subsequent calls once it was compiled for a certain vocab size). I just forgot to get rid of that last comment. Sorry about that!

My memory from this is that I stopped trying that hcat approach because it was indeed nice and quick for the smaller glove files like you said, but it seemed like it was taking much too long with the large files (as in, minutes longer to hcat vectors compared to use an array from the start, at least on my laptop).

In any case, I'd be very happy to update the PR this evening to remove the unnecessary comment at the very least. I'll revisit the hcat idea as well and see if I have any better luck making it fast.

Thanks again!